### PR TITLE
Fixes #27186 - Update katello-remove to include /var/cache/pulp

### PR DESCRIPTION
**Pull request by chris1984**
Also Added/Removed things based on nightly and 6.6 and sorted the
packages and content for easier readability

### CODE REVIEW

**Comment by chris1984** [comment]
Proof that mount point exists:

```bash

└─/var/cache/pulp                     /dev/mapper/centos_remove-var_cache_pulp
                                                                   xfs        rw,relatime,seclabel,attr2,inode64,noquota[root@remove ~]# ll /var/cache/pulp/
total 0
drwxr-xr-x. 2 apache apache 6 Jun 24 09:41 reserved_resource_worker-0@remove.katello.lan
drwxr-xr-x. 2 apache apache 6 Jun 24 09:31 reserved_resource_worker-1@remove.katello.lan
drwxr-xr-x. 2 apache apache 6 Jun 24 09:31 resource_manager@remove.katello.lan

```

Removal with patch:

```bash

Removing Databases and Pulp storage on Filesystem

 /var/cache/pulp is part of a mounted filesystem, skipping. You will need to remove /var/cache/pulp manually.

Finished

```

File: packages/katello/katello/katello-remove [hunk 1]
```diff
@@ -68,43 +74,43 @@ LOG_FILES = [
 ]
 
 RPMS = [
-  'puppetlabs-release',
-  'foreman-release',
-  'foreman-client',
-  'foreman-proxy',
-  'candlepin',
-  'katello',
+  '$HOSTNAME',
```
Comment (ekohl) [inline packages/katello/katello/katello-remove]: Does this work? By default variables within single quotes are not interpolated

Comment (chris1984) [reply packages/katello/katello/katello-remove]: @ekohl it works, it looks like it has been that way for a while, I was surprised too. I did a quick test on up/downstream to confirm. Here are the findings showing it works:

Before on upstream:
```bash
[root@centos7-katello-nightly ~]# rpm -qa | grep nightly
centos7-katello-nightly.area51.example.com-qpid-router-server-1.0-1.noarch
katello-repos-3.13.0-0.1.nightly.el7.noarch
centos7-katello-nightly.area51.example.com-tomcat-1.0-1.noarch
centos7-katello-nightly.area51.example.com-foreman-proxy-1.0-1.noarch
centos7-katello-nightly.area51.example.com-puppet-client-1.0-1.noarch
centos7-katello-nightly.area51.example.com-qpid-broker-1.0-1.noarch
centos7-katello-nightly.area51.example.com-apache-1.0-1.noarch
centos7-katello-nightly.area51.example.com-foreman-proxy-client-1.0-1.noarch
centos7-katello-nightly.area51.example.com-qpid-client-cert-1.0-1.noarch
centos7-katello-nightly.area51.example.com-foreman-client-1.0-1.noarch
centos7-katello-nightly.area51.example.com-qpid-router-client-1.0-1.noarch
```
After on upstream:
```bash
[root@centos7-katello-nightly sbin]# updatedb
[root@centos7-katello-nightly sbin]# rpm -qa | grep nightly
```
Before on downstream:
```bash
[root@sat-6-6-qa-rhel7 ~]# rpm -qa | grep area51
sat-6-6-qa-rhel7.area51.example.com-qpid-broker-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-apache-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-foreman-proxy-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-qpid-router-client-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-qpid-client-cert-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-foreman-client-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-qpid-router-server-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-foreman-proxy-client-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-tomcat-1.0-1.noarch
sat-6-6-qa-rhel7.area51.example.com-puppet-client-1.0-1.noarch
```
After on downstream:
```bash
[root@sat-6-6-qa-rhel7 ~]# updatedb
[root@sat-6-6-qa-rhel7 ~]# rpm -qa | grep area51
```

**Comment by chris1984** [comment]
@ekohl updated when you get time :watch:

File: packages/katello/katello/katello-remove [hunk 2]
```diff
@@ -68,43 +74,43 @@ LOG_FILES = [
...
+  '^rubygem',
+  '^tfm',
+  'candlepin',
+  'foreman-client',
+  'foreman-proxy',
+  'foreman-release',
   'httpd',
+  'katello',
+  'mongo',
+  'postgre',
```
Comment (ekohl) [inline packages/katello/katello/katello-remove]: Why `postgre`?

File: packages/katello/katello/katello-remove [hunk 3]
```diff
@@ -68,43 +74,43 @@ LOG_FILES = [
...
   'puppet',
-  'tomcat',
-  'squid'
+  'puppetlabs-release',
```
Comment (ekohl) [inline packages/katello/katello/katello-remove]: Why do you want to remove this? Would you also remove epel-release?

File: packages/katello/katello/katello-remove [hunk 4]
```diff
@@ -68,43 +74,43 @@ LOG_FILES = [
...
-  '$HOSTNAME',
-  'saslwrapper',
-  'ruby',
-  'rubygems',
+  '^python-webpy',
+  '^qpid',
+  '^rubygem',
+  '^tfm',
+  'candlepin',
+  'foreman-client',
```
Comment (ekohl) [inline packages/katello/katello/katello-remove]: If you have `^foreman`, aren't these just a subset?

File: packages/katello/katello/katello-remove [hunk 5]
```diff
@@ -120,38 +126,41 @@ CERT_FILES = [
...
   '/var/lib/katello',
   '/var/lib/puppet',
   '/var/lib/qpidd',
   '/var/lib/tomcat',
   '/var/opt/theforeman',
+  '/var/spool/squid',
   '/var/tmp/foreman-proxy'
 ]
 
 MOUNTS = [
```
Comment (ekohl) [inline packages/katello/katello/katello-remove]: Why do we even track mounts? Don't you just want to check if a `content` is a mount, then remove `#{content}/*`

**Comment by chris1984** [comment]
@ekohl updated, output of tests:

Downstream:
```bash
├─/var/cache/pulp                     /dev/mapper/centos-var_cache_pulp
                                                 xfs        rw,relatime,seclabel,attr2,inode64,noquota

Starting removal

Stopping services

Removing RPMs

Cleaning up configuration files

Cleaning up log files

Cleaning up certs

Cleaning up content

 /var/cache/pulp is part of a mounted filesystem, skipping. You will need to remove /var/cache/pulp manually.

Finished
```
Upstream:
```bash
└─/usr/share/foreman                  /dev/mapper/centos_remove00-var_cache_pulp
                                                 xfs        rw,relatime,seclabel,attr2,inode64,noquota
												 
Starting removal

Stopping services

Removing RPMs

Cleaning up configuration files

Cleaning up log files

Cleaning up certs

Cleaning up content

 /usr/share/foreman is part of a mounted filesystem, skipping. You will need to remove /usr/share/foreman manually.

Finished
```

File: packages/katello/katello/katello-remove [hunk 6]
```diff
@@ -44,10 +45,15 @@ CONFIG_FILES = [
   '/etc/puppetlabs',
   '/etc/qpid-dispatch',
   '/etc/qpid',
+  '/etc/smart_proxy_dynflow_core',
   '/etc/squid',
   '/etc/sudoers.d/foreman-proxy',
-  '/etc/sysconfig/foreman.rpmsave',
-  '/etc/sysconfig/puppetserver.rpmsave',
+  '/etc/systemd/system/multi-user.target.wants/pulp_celerybeat.service',
```
Comment (ekohl) [inline packages/katello/katello/katello-remove]: Why are these needed? Doesn't the Pulp RPM removal clean this up? If not, then that's a bug in itself IMHO.

**Comment by chris1984** [comment]
@ekohl here is the pulp issue:

https://pulp.plan.io/issues/5195

File: packages/katello/katello/katello-remove (hunk 6 above)
Comment (ekohl) [reply packages/katello/katello/katello-remove]: What I meant was:
```suggestion
  # https://pulp.plan.io/issues/5195
  '/etc/systemd/system/multi-user.target.wants/pulp_celerybeat.service',
```

**Comment by chris1984** [comment]
@ekohl updated :)
